### DISCUSSION TITLE
'Invalid audience' from Strapi Microsoft provider callback request.

### DISCUSSION THREAD

**Discussion by andrewxt** [strapi/strapi#7445] [Legacy Topics]
I am trying to configure authentication using Microsoft in an Angular app. I am able to authenticate in the app successfully. When I send the token I receive to 'http://localhost:1337/auth/microsoft/callback?access_token=${token}' I get a response back saying 'Access token validation failure. Invalid audience.' I checked the token I am sending in jwt.io and the audience matches the client id configured in the Strapi admin config for the Microsoft provider. Am I missing a step?

**Comment by andrewxt** [comment]
My request looks like this and is fired after auth is complete and the token is set. 

`      let token = window.localStorage.getItem('msal.idtoken')
      this.http.get(`http://localhost:1337/auth/microsoft/callback?access_token=${token}`)
        .subscribe((res) => {
          console.log(res);
        });`

**Reply by derrickmehaffy** [reply]
@petersg83 Do you have suggestions here, I'm not familiar with the Microsoft auth API and in my own testing I've never seen the `Access token validation failure. Invalid audience` issue. Assuming there is some breakdown of what we are requesting? I assume it relates to calling the auth API from localhost and Microsoft may require HTTPS.

**Reply by petersg83** [reply]
Hi,
I've never seen that before neither. Microsoft was working fine on my side few days ago. So no clue for the moment :/

**Reply by andrewxt** [reply]
@petersg83 Do you have an idea of what I might be missing in my request or something that might be misconfigured?

**Reply by petersg83** [reply]
If you want to dig a bit in the code, the error probably comes from this part of the code: https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/services/Providers.js#L253
It is where i would start digging

**Comment by biapar** [comment]
Which token Do you sent? Bearer?
